Baikal/PostEffects/post_effect.cpp

```diff
@@ -0,0 +1,127 @@
+/**********************************************************************
+Copyright (c) 2016 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+********************************************************************/
+
+#include "post_effect.h"
+
+namespace Baikal
+{
+
+    PostEffect::Param::Param(std::uint32_t value)
+            : m_type(ParamType::kUintVal)
+    {
+        m_value.uint_value = value;
+    }
+
+    PostEffect::Param::Param(float value)
+                      : m_type(ParamType::kFloatVal)
+    {
+        m_value.float_value = value;
+    }
+
+    PostEffect::Param::Param(RadeonRays::float4 const& value)
+            : m_type(ParamType::kFloat4Val)
+    {
+        m_value.float4_value = value;
+    }
+
+    PostEffect::Param::Param(std::string const& value)
+            : m_type(ParamType::kStringVal)
+    {
+        m_value.str_value = value;
+    }
+
+    PostEffect::ParamType PostEffect::Param::GetType() const
+    {
+        return m_type;
+    }
+
+    float PostEffect::Param::GetFloatVal() const
+    {
+        if (m_type != ParamType::kFloatVal)
```

Comment:
The messages are always the same. I'd propose private `void AssertType(ParamType) const` for checking.

Reply:
Done